File: examples/nodejs/today_orders.js
const { Config, TradeContext, OAuth } = require("longport");

async function main() {
  const oauth = new OAuth("your-client-id");
  const token = await oauth.authorize((url) => {
    console.log(url);
  });
  let config = Config.fromOauth("your-client-id", token.accessToken);
  let ctx = await TradeContext.new(config);
  let resp = await ctx.todayOrders();
  for (let obj of resp) {
    console.log(obj.toString());
  }
}

Promise.all([main()]).catch((err) => console.error(err));

File: examples/python/today_orders.py
import asyncio

from longport.openapi import TradeContext, Config, OAuth, OAuthToken


async def get_config() -> Config:
    try:
        token = OAuthToken.load()
    except Exception:
        oauth = OAuth("your-client-id")
        token = await oauth.authorize(lambda url: print(f"Open this URL to authorize: {url}"))
        token.save()
    return Config.from_oauth(token)


config = asyncio.run(get_config())
ctx = TradeContext(config)
resp = ctx.today_orders()
print(resp)
